From 707fbfb32f82f253a1792347ce76bca98679c8d8 Mon Sep 17 00:00:00 2001
From: Nobuhiro Iwamatsu <iwamatsu@nigauri.org>
Date: Sun, 28 Aug 2022 09:06:14 +0900
Subject: [PATCH 1/2] Fix static link

Signed-off-by: Nobuhiro Iwamatsu <iwamatsu@nigauri.org>
---
 programs/Makefile | 6 +++---
 tests/Makefile    | 4 +++-
 2 files changed, 6 insertions(+), 4 deletions(-)

diff --git a/programs/Makefile b/programs/Makefile
index ace0d03..fc4d6f1 100644
--- a/programs/Makefile
+++ b/programs/Makefile
@@ -91,10 +91,10 @@ lz4-exe.o: lz4-exe.rc
 	$(WINDRES) -i lz4-exe.rc -o lz4-exe.o
 
 lz4: $(OBJFILES) lz4-exe.o
-	$(CC) $(FLAGS) $^ -o $@$(EXT)
+	$(CC) $(FLAGS) $^ -o $@$(EXT) -L ../lib -llz4
 else
 lz4: $(OBJFILES)
-	$(CC) $(FLAGS) $(OBJFILES) -o $@$(EXT) $(LDLIBS)
+	$(CC) $(FLAGS) $(OBJFILES) -o $@$(EXT) $(LDLIBS) -L ../lib -llz4
 endif
 
 .PHONY: lz4-release
@@ -118,7 +118,7 @@ lz4c: lz4
 
 lz4c32: CFLAGS += -m32
 lz4c32 : $(SRCFILES)
-	$(CC) $(FLAGS) $^ -o $@$(EXT)
+	$(CC) $(FLAGS) $^ -o $@$(EXT) -L ../lib -llz4
 
 lz4.1: lz4.1.md $(LIBVER_SRC)
 	cat $< | $(MD2ROFF) $(MD2ROFF_FLAGS) | $(SED) -n '/^\.\\\".*/!p' > $@
diff --git a/tests/Makefile b/tests/Makefile
index 93a5581..b48070a 100644
--- a/tests/Makefile
+++ b/tests/Makefile
@@ -43,6 +43,8 @@ CFLAGS  += $(DEBUGFLAGS) $(MOREFLAGS)
 CPPFLAGS+= -I$(LZ4DIR) -I$(PRGDIR) -DXXH_NAMESPACE=LZ4_
 FLAGS    = $(CFLAGS) $(CPPFLAGS) $(LDFLAGS)
 
+export LD_LIBRARY_PATH += $(LZ4DIR)
+
 include ../Makefile.inc
 
 LZ4 := $(PRGDIR)/lz4$(EXT)
@@ -62,7 +64,7 @@ all32: CFLAGS+=-m32
 all32: all
 
 lz4:
-	$(MAKE) -C $(PRGDIR) $@ CFLAGS="$(CFLAGS)"
+	#$(MAKE) -C $(PRGDIR) $@ CFLAGS="$(CFLAGS)"
 
 lib liblz4.pc:
 	$(MAKE) -C $(LZ4DIR) $@ CFLAGS="$(CFLAGS)"
-- 
2.36.1

