Description: Do not use bundled xxHash library.
Origin: Sutou Kouhei <kou@clear-code.com>
Bug-Debian: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=1073192
Forwarded: not-needed
Author: Kentaro Hayashi <kenhys@xdump.org>
Last-Update: 2025-03-07

Signed-off-by: Kentaro Hayashi <kenhys@gmail.com>
Signed-off-by: Nobuhiro Iwamatsu <iwamatsu@debian.org>
---

diff --git a/build/cmake/CMakeLists.txt b/build/cmake/CMakeLists.txt
index c372733a..48a849f0 100644
--- a/build/cmake/CMakeLists.txt
+++ b/build/cmake/CMakeLists.txt
@@ -77,15 +77,19 @@ include_directories("${LZ4_LIB_SOURCE_DIR}")
 # CLI sources
 file(GLOB LZ4_SOURCES
      "${LZ4_LIB_SOURCE_DIR}/*.c")
+list(REMOVE_ITEM LZ4_SOURCES "${LZ4_LIB_SOURCE_DIR}/xxhash.c")
 file(GLOB LZ4_CLI_SOURCES
      "${LZ4_PROG_SOURCE_DIR}/*.c")
-list(APPEND LZ4_CLI_SOURCES ${LZ4_SOURCES}) # LZ4_CLI always use liblz4 sources directly.
+# list(APPEND LZ4_CLI_SOURCES ${LZ4_SOURCES}) # LZ4_CLI always use liblz4 sources directly.
 
 # Whether to use position independent code for the static library.  If
 # we're building a shared library this is ignored and PIC is always
 # used.
 option(LZ4_POSITION_INDEPENDENT_LIB "Use position independent code for static library (if applicable)" ON)
 
+find_package(PkgConfig REQUIRED)
+pkg_check_modules(XXHASH REQUIRED libxxhash)
+
 # liblz4
 include(GNUInstallDirs)
 set(LZ4_LIBRARIES_BUILT)
@@ -103,6 +107,7 @@ if(BUILD_SHARED_LIBS)
       LZ4_DLL_EXPORT=1)
   endif()
   list(APPEND LZ4_LIBRARIES_BUILT lz4_shared)
+  target_link_libraries(lz4_shared ${XXHASH_LIBRARIES})
 endif()
 if(BUILD_STATIC_LIBS)
   set(STATIC_LIB_NAME lz4)
@@ -117,6 +122,7 @@ if(BUILD_STATIC_LIBS)
     OUTPUT_NAME ${STATIC_LIB_NAME}
     POSITION_INDEPENDENT_CODE ${LZ4_POSITION_INDEPENDENT_LIB})
   list(APPEND LZ4_LIBRARIES_BUILT lz4_static)
+  target_link_libraries(lz4_static ${XXHASH_STATIC_LIBRARIES})
 endif()
 # Add unified target.
 add_library(lz4 INTERFACE)
@@ -127,21 +133,12 @@ else()
   target_link_libraries(lz4 INTERFACE lz4_static)
 endif()
 
-# xxhash namespace
-if(BUILD_SHARED_LIBS)
-  target_compile_definitions(lz4_shared PRIVATE
-    XXH_NAMESPACE=LZ4_)
-endif()
-if(BUILD_STATIC_LIBS)
-  target_compile_definitions(lz4_static PRIVATE
-    XXH_NAMESPACE=LZ4_)
-endif()
-
 # lz4
 if (LZ4_BUILD_CLI)
   set(LZ4_PROGRAMS_BUILT lz4cli)
   add_executable(lz4cli ${LZ4_CLI_SOURCES})
   set_target_properties(lz4cli PROPERTIES OUTPUT_NAME lz4)
+  target_link_libraries(lz4cli lz4 ${XXHASH_LIBRARIES})
 endif()
 
 # Extra warning flags
diff --git a/lib/liblz4.pc.in b/lib/liblz4.pc.in
index ed52214f..cbbc7cd3 100644
--- a/lib/liblz4.pc.in
+++ b/lib/liblz4.pc.in
@@ -10,5 +10,5 @@ Name: lz4
 Description: extremely fast lossless compression algorithm library
 URL: http://www.lz4.org/
 Version: @VERSION@
-Libs: -L${libdir} -llz4
+Libs: -L${libdir} -llz4 -lxxhash
 Cflags: -I${includedir}
